src/Libraries/Nop.Services/Catalog/ProductService.cs

```diff
@@ -1766,6 +1767,17 @@ public virtual async Task<string[]> GetNotExistingProductsAsync(string[] product
             return queryFilter.Except(filter).ToArray();
         }
 
+        public async Task<Product> GetByTag(string tag)
+        {
+            var tagEntity = await _productTagRepository.Table.FirstOrDefaultAsync(x => x.Name == tag);
+            if (tagEntity is null)
+            {
+                throw new ArgumentException($"Can not find {tag} tag");
+            }
+            var productId = _productTagMappingRepository.Table.FirstOrDefault(x => x.ProductTagId == tagEntity.Id)?.ProductId;
+            return productId == null ? null : await GetProductByIdAsync(Convert.ToInt32(productId));
```

Comment:
why this convert.ToINt32 is reuqired?

Reply:
productId was nullable and the method GetProductById requires a non nullable int, so it was the solution. Maybe better way to fix this was `!` sign.